core/Pos/Aeson/Crypto.hs

```diff
@@ -45,7 +55,39 @@ instance FromJSON PublicKey where
 instance FromJSON (Signature w) where
     parseJSON v = parseJSON v >>= eitherMsgFail "Signature" . parseFullSignature
 
+instance ToJSON (Signature w) where
+    toJSON = toJSON . sformat fullSignatureHexF
+
 instance FromJSON (ProxyCert w) where
     parseJSON v = parseJSON v >>= eitherMsgFail "Signature" . parseFullProxyCert
 
 deriveJSON defaultOptions ''ProxySecretKey
+
+-- All these instances below needed for wallet
+
+instance FromJSON Ed25519.PublicKey where
+    parseJSON v = Ed25519.PublicKey . getJsonByteString <$> parseJSON v
+
+instance ToJSON Ed25519.PublicKey where
+    toJSON = toJSON . JsonByteString . Ed25519.openPublicKey
+
+instance FromJSON Ed25519.Signature where
+    parseJSON v = Ed25519.Signature . getJsonByteString <$> parseJSON v
+
+instance ToJSON Ed25519.Signature where
+    toJSON = toJSON . JsonByteString . Ed25519.unSignature
+
+instance FromJSON CC.XSignature where
+    parseJSON v = either (error . toText) identity . CC.xsignature . getJsonByteString <$> parseJSON v
+
+instance ToJSON CC.XSignature where
+    toJSON = toJSON . JsonByteString . CC.unXSignature
+
+instance FromJSON HDAddressPayload where
+    parseJSON v = HDAddressPayload . getJsonByteString <$> parseJSON v
+
+instance ToJSON HDAddressPayload where
+    toJSON = toJSON . JsonByteString . getHDAddressPayload
+
+deriveJSON defaultOptions ''RedeemPublicKey
```

Comment:
We are good, see [here](https://github.com/input-output-hk/cardano-sl/blob/master-release-candidate/core/Pos/Aeson/Crypto.hs#L65-L86).